### PR TITLE
move two methods from download_strategy to utils/github

### PR DESCRIPTION
**Pull request by hyuraku**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
move two methods (`github_last_commit` and `multiple_short_commits_exist` ) from `download_storategy` to `utils/github` because of `# TODO`.

### CODE REVIEW

File: Library/Homebrew/utils/github.rb [hunk 1]
```diff
@@ -666,4 +666,32 @@ def pull_request_labels(user, repo, pr)
...
+    version.update_commit(commit) if commit
+    commit
+  end
+
+  def multiple_short_commits_exist?(user, repo, commit)
+    return if Homebrew::EnvConfig.no_github_api?
+
+    output, _, status = curl_output(
+      "--silent", "--head", "--location",
+      "-H", "Accept: application/vnd.github.sha",
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/github.rb:691]:
```suggestion
      "--header", "Accept: application/vnd.github.sha",
```

File: Library/Homebrew/utils/github.rb [hunk 2]
```diff
@@ -688,10 +690,14 @@ def multiple_short_commits_exist?(user, repo, commit)
...
+      "--header", "Accept: application/vnd.github.sha",
       "https://api.github.com/repos/#{user}/#{repo}/commits/#{commit}"
     )
 
-    !(status.success? && output && output[/^Status: (200)/, 1] == "200")
+    return true if status.success?
+    return true unless output
+    return true if output[/^Status: (200)/, 1] != "200"
+
+    false
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/github.rb:701]:
```suggestion
    return true unless status.success?
    return true if output.blank?
    
    output[/^Status: (200)/, 1] != "200"
```

I think? Please disagree with me if you don't think I'm right as I'm not 100% sure.

Comment (hyuraku) [reply Library/Homebrew/utils/github.rb:701]: I agree with the refactor code; I was wrong and will fix it.

File: Library/Homebrew/utils/github.rb [hunk 3]
```diff
@@ -666,4 +666,37 @@ def pull_request_labels(user, repo, pr)
...
     pr_data["labels"].map { |label| label["name"] }
   end
+
+  def last_commit(user, repo, ref)
+    return if Homebrew::EnvConfig.no_github_api?
+
+    output, _, status = curl_output(
+      "--silent", "--head", "--location",
+      "--header", "Accept: application/vnd.github.sha",
+      "https://api.github.com/repos/#{user}/#{repo}/commits/#{ref}"
```
Comment (Bo98) [inline Library/Homebrew/utils/github.rb:676]:
```suggestion
      url_to("repos", user, repo, "commits", ref)
```

(might need a `.to_s` - not sure)

File: Library/Homebrew/utils/github.rb [hunk 4]
```diff
@@ -666,4 +666,37 @@ def pull_request_labels(user, repo, pr)
...
+    commit
+  end
+
+  def multiple_short_commits_exist?(user, repo, commit)
+    return if Homebrew::EnvConfig.no_github_api?
+
+    output, _, status = curl_output(
+      "--silent", "--head", "--location",
+      "--header", "Accept: application/vnd.github.sha",
+      "https://api.github.com/repos/#{user}/#{repo}/commits/#{commit}"
```
Comment (Bo98) [inline Library/Homebrew/utils/github.rb:694]:
```suggestion
      url_to("repos", user, repo, "commits", commit)
```

(might need a `.to_s` - not sure)

File: Library/Homebrew/utils/github.rb [hunk 5]
```diff
@@ -666,4 +666,37 @@ def pull_request_labels(user, repo, pr)
...
+    return if Homebrew::EnvConfig.no_github_api?
+
+    output, _, status = curl_output(
+      "--silent", "--head", "--location",
+      "--header", "Accept: application/vnd.github.sha",
+      url_to("repos", user, repo, "commits", commit).to_s
+    )
+
+    return true if status.success?
+    return true unless output
```
Comment (MikeMcQuaid) [inline Library/Homebrew/utils/github.rb:698]:
```suggestion
    return true unless status.success?
    return true if output.blank?
```

I'm pretty sure?

Comment (hyuraku) [reply Library/Homebrew/utils/github.rb:698]: It's right, the original code `!(status.success? && output && output[/^Status: (200)/, 1] == "200")` is equal to `!status.success? || !output || output[/^Status: (200)/, 1] != "200"`, therefore `return true unless status.success?` is right.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks again @hyuraku!

**Pull request merged**